appclient/src/main/java/org/jboss/as/appclient/service/ApplicationClientDeploymentService.java

```diff
@@ -32,68 +32,66 @@
+import java.util.function.Supplier;
 
-import org.jboss.as.controller.ModelController;
-import org.jboss.as.controller.client.ModelControllerClient;
+import org.jboss.as.controller.LocalModelControllerClient;
+import org.jboss.as.controller.ModelControllerClientFactory;
 import org.jboss.as.controller.client.Operation;
 import org.jboss.as.controller.client.OperationBuilder;
 import org.jboss.as.controller.operations.common.Util;
 import org.jboss.as.server.deployment.DeploymentAddHandler;
 import org.jboss.as.server.deployment.DeploymentDeployHandler;
 import org.jboss.dmr.ModelNode;
-import org.jboss.msc.service.Service;
-import org.jboss.msc.service.ServiceName;
+import org.jboss.msc.Service;
 import org.jboss.msc.service.StartContext;
-import org.jboss.msc.service.StartException;
 import org.jboss.msc.service.StopContext;
-import org.jboss.msc.value.InjectedValue;
 
 /**
  * Service responsible for deploying the application client that was specified on the command line
  *
  * @author Stuart Douglas
  */
-public class ApplicationClientDeploymentService implements Service<ApplicationClientDeploymentService> {
+public class ApplicationClientDeploymentService implements Service {
 
 
-    public static final ServiceName SERVICE_NAME = ServiceName.JBOSS.append("appClientDeploymentService");
-
     private final File path;
-    private ModelControllerClient controllerClient;
-    private final InjectedValue<ModelController> controllerValue = new InjectedValue<ModelController>();
+    private final Consumer<ApplicationClientDeploymentService> consumer;
+    private final Supplier<ModelControllerClientFactory> clientFactorySupplier;
+    private final Supplier<ExecutorService> executorServiceSupplier;
     private final CountDownLatch deploymentCompleteLatch = new CountDownLatch(1);
 
 
-    public ApplicationClientDeploymentService(final File path) {
+    public ApplicationClientDeploymentService(final Consumer<ApplicationClientDeploymentService> consumer,
+                                              final File path,
+                                              final Supplier<ModelControllerClientFactory> clientFactorySupplier,
+                                              final Supplier<ExecutorService> executorServiceSupplier) {
+        this.consumer = consumer;
         this.path = path;
+        this.clientFactorySupplier = clientFactorySupplier;
+        this.executorServiceSupplier = executorServiceSupplier;
     }
 
     @Override
-    public synchronized void start(final StartContext context) throws StartException {
-        controllerClient = controllerValue.getValue().createClient(Executors.newSingleThreadExecutor());
-
+    public synchronized void start(final StartContext context) {
         final DeployTask task = new DeployTask();
+        // TODO use executorServiceSupplier
         Thread thread = new Thread(new DeploymentTask(new OperationBuilder(task.getUpdate()).build()));
         thread.start();
```

Comment:
I don't know. It seems reasonable but OTOH Stuart wrote this and he understood these things, so I don't want to assume it should change.

Reply:
@pferraro It can't do that. What the task does is execute a management op (deploy the app). If it does that in the context of the existing op that adds this service that op will hang.